sources/modules/program-design/iterators.md

```diff
@@ -0,0 +1,138 @@
+## Program Design: iterators {#it}
+
+_Skeleton descriptions are typeset in italic text,_
+_so please don't remove these descriptions when editing the topic._
+
+### Overview
+
+_Provides a short natural language abstract of the module’s contents._
+_Specifies the different levels of teaching._
+
+------------------------------------------------------------------------
+Level             Objective
+----------------- ------------------------------------------------------
+Foundational      Using iterators
+
+Main              Writing iterators
+
+Advanced          ---
```

Comment:
How about "endowing iterators with execution policies" under advanced?

Reply:
I do not mind this being added, but I would change the wording because I doubt "endow" is a common enough verb for every non-native speaker.  I do not have a better replacement on the top of my head, maybe `extend`, since it's more used in programming circles?  It could be a bit too overloaded though.